lib/plugins/aws/customResources/resources/utils.js

```diff
@@ -56,13 +56,22 @@ function response(event, context, status, data = {}, err) {
 }
 
 function getLambdaArn(region, accountId, functionName) {
-  return `arn:aws:lambda:${region}:${accountId}:function:${functionName}`;
+  let awsNode = 'aws';
+  if (region === 'us-gov-west-1' || region === 'us-gov-east-1') {
+    awsNode = 'aws-us-gov';
+  }
+  return `arn:${awsNode}:lambda:${region}:${accountId}:function:${functionName}`;
 }
 
 function getEnvironment(context) {
```

Comment:
Perhaps replace this function to gather the partition in the regex

```javascript
function getEnvironment(context) {
  const arn = context.invokedFunctionArn.match(
    /^arn:(aws[\w-]*).*:lambda:(\w+-\w+-\d+):(\d+):function:(.*)$/
  );
  return {
    LambdaArn: arn[0],
    Partition: arn[1],
    Region: arn[2],
    AccountId: arn[3],
    LambdaName: arn[4],
  };
}
```

Reply:
Good suggestion it's `getEnvironment` which should resolve partition

Reply:
Right, that's exactly what I just attempted to do.  However, the region for GovCloud has a different format: `us-gov-west-1` and `us-gov-east-1` that is incompatible with the existing regex that extracts region.  Any suggestions?

This works for govcloud, but not for regular public regions
```js
/^arn:(aws[\w-]*).*:lambda:(\w+-\w+-\w+-\d+):(\d+):function:(.*)$/
```

This works for public regions, but not for govcloud:
```js
/^arn:(aws[\w-]*).*:lambda:(\w+-\w+-\d+):(\d+):function:(.*)$/
```

Reply:
Oh good point!

In that case I think adjusting the regex in this way should work

`/^arn:(aws[\w-]*).*:lambda:([\w+-]{2,}\d+):(\d+):function:(.*)$/`